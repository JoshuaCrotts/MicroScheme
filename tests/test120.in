;
; Simple "merge" procedure.
;

(define (merge l r)
    (cond 
        ((null? l) r)
        ((null? r) l)
        (else 
            (let ((a (car l)) (b (car r)))
                (if (< a b)
                    (cons a (merge (cdr l) r))
                    (cons b (merge (cdr r) l)))))))

(merge '(1 7 11 24) '(3 4 5 12))
