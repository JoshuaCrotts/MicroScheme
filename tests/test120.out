(1 3 4 5 7 11 12 24)
